guides/source/asset_pipeline.md

```diff
@@ -8,479 +8,616 @@ This guide covers the asset pipeline.
+   browsers, reducing the need for transpilation and minification.
 
-You can find detailed usage instructions on the Gem homepage, but it's important to understand the basics of `importmap-rails`.
+Therefore, in Rails 8, the asset pipeline powered by Propshaft, no longer
+includes transpilation, bundling, or compression by default. However,
+fingerprinting remains an integral part of Propshaft. You can learn more about
+it below.
 
-### How it works
+### Transpilation ❌
 
-Import maps are essentially a string substitution for what is referred to as "bare module specifiers". They allow you to standardize the names of JavaScript module imports.
+Transpilation involves converting code from one language or format to another.
 
-Take for example such an import definition, it won't work without an import map:
+For example, converting CoffeeScript to JavaScript.
 
 ```javascript
-import React from "react"
+alert "Hello, world!"
 ```
 
-You would have to define it like this to make it work:
+After transpilation, this code becomes:
 
 ```javascript
-import React from "https://ga.jspm.io/npm:react@17.0.2/index.js"
+alert("Hello, world!");
 ```
 
-Here comes the import map, we define the `react` name to be pinned to the `https://ga.jspm.io/npm:react@17.0.2/index.js` address. With such information our browser accepts the simplified `import React from "react"` definition. Think of import map as about an alias for the library source address.
+In the past, pre-processors like [Sass](https://sass-lang.com/) and
+[Less](https://lesscss.org/) were essential for CSS features such as variables
+and nesting. Today, modern CSS supports these natively, reducing the need for
+transpilation.
 
-### Usage
+### Bundling ❌
 
-With `importmap-rails` you create the importmap configuration file pinning the library path to a name:
+Bundling combines multiple files into one to reduce the number of HTTP requests
+a browser needs to make to render a page.
 
-```ruby
-# config/importmap.rb
-pin "application"
-pin "react", to: "https://ga.jspm.io/npm:react@17.0.2/index.js"
-```
-
-All of the configured import maps should be attached in `<head>` element of your application by adding  `<%= javascript_importmap_tags %>` . The `javascript_importmap_tags` renders a bunch of scripts in the `head` element:
+For example, if your application has three JavaScript files:
+- menu.js
+- cart.js
+- checkout.js
 
-- JSON with all configured import maps:
+Bundling will merge these into a single application.js file.
 
-```html
-<script type="importmap">
-{
-  "imports": {
-    "application": "/assets/application-39f16dc3f3....js"
-    "react": "https://ga.jspm.io/npm:react@17.0.2/index.js"
-  }
-}
-</script>
-```
-
-- Entrypoint for loading JavaScript from `app/javascript/application.js`:
-
-```html
-<script type="module">import "application"</script>
+```javascript
+// Contents of menu.js, cart.js, and checkout.js are combined here
```

Comment:
```suggestion
```javascript
// app/javascript/application.js
// Contents of menu.js, cart.js, and checkout.js are combined here
```